### PR TITLE
Hide wind speed and direction

### PR DESCRIPTION
**Pull request by akropp**
Wanted to be able to only show the wind speed (without the direction).  For completeness, added two flags: hide_wind_speed and hide_wind_direction.  They work in concert with show_wind, and default to false (so the existing behavior).

### CODE REVIEW

**Review by decompil3d — CHANGES_REQUESTED** [review]
Thanks so much for your contribution! I like the idea of being able to show only wind speed or direction. I think that we can achieve the same result without adding more flags though.

Can you please refactor to make `show_wind` take these values:

- `true`: show both wind speed and direction
- `false`: show neither speed nor direction
- `'speed'`: show only wind speed
- `'direction'`: show only wind direction

Then, in the editor UI, you can just replace the existing wind switch with a drop-down or similar to choose from these options.

What do you think?

**Comment by akropp** [comment]
Cool.  Good idea.  I didn't really like the separate flags thing either.

--Adam

File: src/weather-bar.ts [hunk 1]
```diff
@@ -69,7 +69,8 @@ export class WeatherBar extends LitElement {
       const skipLabel = (i - 1) % this.label_spacing !== 0;
       const hideHours = this.hide_hours || skipLabel;
       const hideTemperature = this.hide_temperatures || skipLabel;
-      const showWind = this.show_wind && !skipLabel;
+      const showWindSpeed = (this.show_wind == 'true' || this.show_wind == 'speed') && !skipLabel;
+      const showWindDirection = (this.show_wind == 'true' || this.show_wind == 'direction') && !skipLabel;
       const showPrecipitationAmounts = this.show_precipitation_amounts && !skipLabel;
       const { hour, temperature } = this.temperatures[i];
       const { windSpeed, windDirection } = this.wind[i];
```
Comment (decompil3d) [inline src/weather-bar.ts:76]:
```suggestion
      const { windSpeed, windDirection } = this.wind[i];

      const wind: TemplateResult[] = [];
      if (showWindSpeed) wind.push(html`${windSpeed}`);
      if (showWindSpeed && showWindDirection) wind.push(html`<br>`);
      if (showWindDirection) wind.push(html`${windDirection}`);

```

File: src/weather-bar.ts [hunk 2]
```diff
@@ -81,7 +82,13 @@ export class WeatherBar extends LitElement {
...
             <div class="hour">${hideHours ? null : hour}</div>
             <div class="temperature">${hideTemperature ? null : html`${temperature}&deg;`}</div>
-            <div class="wind">${showWind ? html`${windSpeed}<br>${windDirection}` : null }</div>
+            <div class="wind">${showWindSpeed && showWindDirection 
+              ? html`${windSpeed}<br>${windDirection}` 
+              : showWindSpeed
+                ? html`${windSpeed}` 
+                : showWindDirection 
+                  ? html`${windDirection}` 
+                  : null }</div>
```
Comment (decompil3d) [inline src/weather-bar.ts:91]:
```suggestion
            <div class="wind">${wind}</div>
```

**Review by decompil3d — CHANGES_REQUESTED** [review]
Played with it a little and I do see the editor behavior you described. However, that same behavior happens on the `entity` field and that logic comes from the base custom card template repo. So, I'm inclined to just let it be as-is. If someone really wants to manage their cards in YAML and needs to mess with this field, then they can either use the dashboard-level YAML editor, or they can compose elsewhere and paste in to avoid the partial text validation problem.

Just a small refactor request for how you render the wind elements and I think we'll be good to go.

**Comment by akropp** [comment]
Sounds good.

**Review by decompil3d — APPROVED** [review]

**Pull request merged**